explorer/explorer.go

```diff
@@ -479,33 +454,57 @@ func (exp *explorerUI) prePopulateChartsData() {
 		return
 	}
 
-	log.Info("Pre-populating the charts data. This may take a minute...")
+	startTime := time.Now()
+
+	log.Info("Pre-populating the charts data")
 	log.Debugf("Retrieving charts data from aux DB.")
-	var err error
-	pgData, err := exp.explorerSource.GetPgChartsData()
-	if dbtypes.IsTimeoutErr(err) {
-		log.Warnf("GetPgChartsData DB timeout: %v", err)
-		return
+
+	chartsData := cacheChartsData.get()
+	pgData := make(map[string]*dbtypes.ChartsData)
+	sqliteData := make(map[string]*dbtypes.ChartsData)
+
+	if chartsData != nil {
+		// Pick all the PgCharts
+		for _, chartName := range dbtypes.PgCharts {
+			dataset := chartsData[chartName]
+			pgData[chartName] = dataset
+		}
+
+		// Pick all the sqliteCharts
+		for _, chartName := range dbtypes.SqliteCharts {
+			dataset := chartsData[chartName]
+			sqliteData[chartName] = dataset
+		}
 	}
+
+	var err error
+	pgData, err = exp.explorerSource.PgChartsData(pgData)
```

Comment:
The `pgData` as a return value is not needs.  The changes to the input map will be visible here in the caller.  This works because `PgChartsData` is not in charge of making the map, just updating it.